otherlibs/stdune/src/io.ml

```diff
@@ -254,42 +254,114 @@ struct
+          let message =
+            Printf.sprintf "%s: Is a directory" (Path.to_string dst)
+          in
+          raise (Sys_error message)
+        | Error `Src_missing ->
+          let message =
+            Printf.sprintf "error: %s: No such file or directory"
+              (Path.to_string src)
+          in
+          raise (Sys_error message)
+        | Ok (fd_src, fd_dst, src_size) ->
+          Exn.protectx (fd_src, fd_dst, src_size)
+            ~f:(fun (src, dst, src_size) -> sendfile ~src ~dst src_size)
+            ~finally:(fun (src, dst, _) ->
+              Unix.close src;
+              Unix.close dst)
+
+    let copyfile ?chmod ~src ~dst () =
+      let src = Path.to_string src in
+      let dst = Path.to_string dst in
+      let src_stats =
+        match Unix.stat src with
+        | exception Unix.Unix_error (Unix.ENOENT, _, _) ->
+          let message =
+            Printf.sprintf "error: %s: No such file or directory" src
+          in
+          raise (Sys_error message)
+        | { st_kind = S_DIR; _ } -> raise (Sys_error "Is a directory")
+        | stats -> stats
+      in
+      (try copyfile src dst with
+      | Unix.Unix_error (Unix.EPERM, "unlink", _) ->
+        let message = Printf.sprintf "%s: Is a directory" dst in
+        raise (Sys_error message)
+      | Unix.Unix_error (Unix.ENOENT, "realpath", _) ->
+        let message =
+          Printf.sprintf "error: %s: No such file or directory" src
+        in
+        raise (Sys_error message));
+      match chmod with
+      | None -> ()
+      | Some chmod -> src_stats.st_perm |> chmod |> Unix.chmod dst
   end
 
   let copy_file ?chmod ~src ~dst () =
     Exn.protectx (setup_copy ?chmod ~src ~dst ()) ~finally:close_both
       ~f:(fun (ic, oc) -> copy_channels ic oc)
 
   let copy_file =
-    match Copyfile.available () with
-    | false -> copy_file
-    | true -> (
-      fun ?chmod ~src ~dst () ->
-        let src = Path.to_string src in
-        let dst = Path.to_string dst in
-        let src_stats =
-          match Unix.stat src with
-          | exception Unix.Unix_error (Unix.ENOENT, _, _) ->
-            let message =
-              Printf.sprintf "error: %s: No such file or directory" src
-            in
-            raise (Sys_error message)
-          | { st_kind = S_DIR; _ } -> raise (Sys_error "Is a directory")
-          | stats -> stats
-        in
-        (try Copyfile.copyfile src dst with
-        | Unix.Unix_error (Unix.EPERM, "unlink", _) ->
-          let message = Printf.sprintf "%s: Is a directory" dst in
-          raise (Sys_error message)
-        | Unix.Unix_error (Unix.ENOENT, "realpath", _) ->
-          let message =
-            Printf.sprintf "error: %s: No such file or directory" src
-          in
-          raise (Sys_error message));
-        match chmod with
-        | None -> ()
-        | Some chmod -> src_stats.st_perm |> chmod |> Unix.chmod dst)
+    match Stdlib.Sys.getenv_opt "DUNE_COPY_FILE_IMPL" with
```

Comment:
Why not use a standard Dune configuration setting, which can be set via `dune-workspace` if needed? Sometimes it's convenient to use a variable, but sometimes not.